### PR TITLE
feat: `--no-git` to `cargo shuttle init`

### LINKED ISSUE #1497: [Feature]: Add `--no-git` to `cargo shuttle init`
### Describe the feature

A flag that disables the initialisation of a git repository during the init command.

### Suggestion or Example of how the feature would be used

`cargo shuttle init --no-git ...`

### Duplicate declaration

- [X] I have searched the issues and this feature has not been requested before.

### PR DESCRIPTION
**Pull request by dhruvdabhi101**
## Description of change

- Added a new property in `InitArgs` struct. 
- Used that new property value and passed to the `generate_project` function
- changed init.rs file so that `--no-git` flag can work

Closes #1497 

## How has this been tested? (if applicable)

> I haven't tested this yet, I just wanted to know if gist of this code looks good or do i need to change anything.

### CODE REVIEW

File: cargo-shuttle/src/args.rs [hunk 1]
```diff
@@ -304,6 +304,9 @@ pub struct InitArgs {
     /// Whether to start the container for this project on Shuttle, and claim the project name
     #[arg(long)]
     pub create_env: bool,
+    /// Whethere to initialize git repository or not during the init command
```
Comment (jonaro00) [inline cargo-shuttle/src/args.rs:307]:
```suggestion
    /// Don't initialize a new git repository
```

**Review by jonaro00 — APPROVED** [review]
The changes are looking good! Since this change is so simple, I would say it could pass without testing at all :sweat_smile: . For the sake of it, you can check it manually with just `cargo run -p cargo-shuttle -- init ...`

**Comment by dhruvdabhi101** [comment]
> The changes are looking good! Since this change is so simple, I would say it could pass without testing at all :sweat_smile: . For the sake of it, you can check it manually with just `cargo run -p cargo-shuttle -- init ...`

sure 😃

**Comment by oddgrd** [comment]
Hey @dhruvdabhi101, thanks for this! It looks like you also need to run a `cargo fmt`.

**Comment by dhruvdabhi101** [comment]
> Hey @dhruvdabhi101, thanks for this! It looks like you also need to run a `cargo fmt`.

Yeah, I'll do that ASAP. : )

**Review by jonaro00 — APPROVED** [review]

**Pull request merged**